### DISCUSSION TITLE
Datadog Metrics

### DISCUSSION THREAD

**Discussion by francoiseric** [artilleryio/artillery#1190] [Artillery]
Hi,

I've been using Artillery for a few weeks and love it. I'm also using the Datadog integration to push metrics in our recurring load tests. 
Here is an example of a nightly build and 3 metrics that are generated
![image](https://user-images.githubusercontent.com/81091386/139311027-d88d3268-4a91-43f7-b952-b02e20c0c54b.png)

As you can see the values are : 
- p99: 1470ms
- p95: 1440ms
- median: 970ms

But when I look in the actual Artillery stats generated after the test, I see the following: 
--------------------------------
Summary report @ 03:47:52(+0000)
--------------------------------

vusers.created_by_name.0: ................................... 228
vusers.created.total: ....................................... 228
vusers.completed: ........................................... 228
vusers.session_length:
  min: ...................................................... 723.6
  max: ...................................................... 4524.6
  median: ................................................... 1064.4
  p95: ...................................................... 2951.9
  p99: ...................................................... 4065.2
http.request_rate: .......................................... 2/sec
http.requests: .............................................. 228
http.codes.200: ............................................. 228
http.responses: ............................................. 228
http.response_time:
  min: ...................................................... 690
  max: ...................................................... 4480
  median: ................................................... 1043.3
  p95: ...................................................... 2893.5
  p99: ...................................................... 4065.2

So I really don't understand where these numbers are coming from.... I've made many tests and they always differ greatly...

Does anybody know what's going on :)

Thanks

**Comment by hassy** [comment]
hi @francoiseric 👋  thanks for the kind words, very glad to hear you're enjoying using Artillery!

This could be due to a couple of things. Do you always run a single instance of Artillery at a time (ie. just one Artillery process)?

Also it's hard to tell from the screenshot, but it looks like there's only one datapoint for each metric there, which should not be the case if the test ran for longer than 10 seconds. What's the query used on that chart?

**Comment by hassy** [comment]
Also, I see "Prod Load Tests" in the title of the chart. 👀  Are you load testing in production? massive kudos and a high five if so! A fellow member of the [production load testing club](https://dev.to/artilleryio/load-testing-in-production-1734)! 😄 

![https://res.cloudinary.com/practicaldev/image/fetch/s--7MrZ4L4W--/c_imagga_scale,f_auto,fl_progressive,h_420,q_auto,w_1000/https://dev-to-uploads.s3.amazonaws.com/uploads/articles/l4ofn49x08n3skcwa5mw.png](https://res.cloudinary.com/practicaldev/image/fetch/s--7MrZ4L4W--/c_imagga_scale,f_auto,fl_progressive,h_420,q_auto,w_1000/https://dev-to-uploads.s3.amazonaws.com/uploads/articles/l4ofn49x08n3skcwa5mw.png)

**Comment by francoiseric** [comment]
Thanks for your answer Hassy!

Here is a screenshot of the Datadog Timeseries Definition:
[image: image.png]

And yes you are right I only have one point for each (and my report is
running for 2 minutes). Whether I choose max, min, avg in the Datadog
Aggregator I have the same value.

Here's my config for the metrics (obfuscated):
 plugins:
    publish-metrics:
      - type: datadog
        prefix: 'ClearD.CDTests.LoadTests.'
        host: <my datadog agent url>
        port: 8126
        tags:
          - "env:prod"
          - "reporterType:datadog-artillery-agent"

**Comment by francoiseric** [comment]
And yes I am launching them in production :)
